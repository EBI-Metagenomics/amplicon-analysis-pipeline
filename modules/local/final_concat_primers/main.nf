
process FINAL_CONCAT_PRIMERS {
    tag "$meta.id"
    label 'light'
    
    input:
    tuple val(meta), path(cat_primers)

    output:
    tuple val(meta), path("*final_concat_primers.fasta"), optional:true, emit: final_concat_primers_out

    """
    cat *concat_primers.fasta > temp_concat_primers.fasta
    if [[ -s temp_concat_primers.fasta ]]; then
        awk '!a[\$0]++' temp_concat_primers.fasta > ${meta.id}_final_concat_primers.fasta
    else
       touch ${meta.id}_final_concat_primers.fasta
    fi
    """

}
