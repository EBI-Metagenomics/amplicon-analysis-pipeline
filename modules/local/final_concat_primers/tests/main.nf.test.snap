{
    "final_concat_primers_out": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "16S-V3-V4",
                    "test_final_concat_primers.fasta:md5,f26cdc22a5231d2dbd0831f1f1983dc6"
                ]
            ]
        ],
        "timestamp": "2023-11-06T14:19:37.304366443"
    }
}